### PR TITLE
Add CPP pragmas for Plutus version

### PR DESCRIPTION
**Pull request by danielfarrelly**
Rebasing #1

### CODE REVIEW

File: ply-plutarch/ply-plutarch.cabal [hunk 1]
```diff
@@ -36,6 +41,7 @@ common common-lang
     BinaryLiterals
     ConstrainedClassMethods
     ConstraintKinds
+    CPP
```
Comment (TotallyNotChase) [inline ply-plutarch/ply-plutarch.cabal:44]: Please don't add CPP as a global extension

File: ply-core/ply-core.cabal [hunk 2]
```diff
@@ -5,6 +5,11 @@ author:             Chase <chase@mlabs.city>
 license:            MIT
 extra-source-files: ../README.md
 
+flag plutus-new
```
Comment (TotallyNotChase) [inline ply-core/ply-core.cabal:8]: let's call this flag `new-ledger-namespace`

File: ply-core/src/Ply.hs [hunk 3]
```diff
@@ -19,11 +19,19 @@ module Ply (
...
   Script (Script),
   Validator (Validator),
  )
+#else
+import Plutus.V1.Ledger.Scripts (
+  MintingPolicy (MintingPolicy),
+  Script (Script),
+  Validator (Validator),
+ )
+#endif
```
Comment (TotallyNotChase) [inline ply-core/src/Ply.hs:34]: A little thing we could do is have a local module called `LedgerExports` where all the necessary ledger stuff are exported and CPP is used. Mostly, it should just export the `PlutusLedgerApi.V1` and a few stuff from `.Scripts` - nothing else will be necessary I presume.

All other modules can use this local module without touching hideous CPP!

**Review by TotallyNotChase — CHANGES_REQUESTED** [review]
Thanks! We can make a few improvements here; it looks ready otherwise

File: ply-core/src/Ply.hs (hunk 3 above)
Comment (danielfarrelly) [reply ply-core/src/Ply.hs]: I've add a module for this, let me know if there's anything else that can be improved here.

Would it be worth having modules for different ledger API versions?

Comment (TotallyNotChase) [reply ply-core/src/Ply.hs]: Yes, in the future - certainly. It's fine for now though. I'll add that during #10

File: ply-core/ply-core.cabal [hunk 4]
```diff
@@ -5,6 +5,11 @@ author:             Chase <chase@mlabs.city>
 license:            MIT
 extra-source-files: ../README.md
 
+flag new-ledger-namespace
+  description: Use the new plutus-ledger-api namespace (0a8b1ad)
+  manual:      True
+  default:     False
```
Comment (TotallyNotChase) [inline ply-core/ply-core.cabal:11]: One final thing: I was thinking about the flag as you mentioned before - and I think the default should actually be `True`. After all, we _want_ people to use the new ledger namespace (and in the future, everyone certainly will). It also makes all of this forwards compatible.

People who _don't_ specify this flag (i.e their projects have already been updated) should never have to change their configs. So we can safely remove this flag in the future.

**Review by TotallyNotChase — CHANGES_REQUESTED** [review]

**Review by TotallyNotChase — APPROVED** [review]
Amazing, thank you again! I'll test some things locally and merge afterwards

**Pull request merged**